### PR TITLE
OSX Code signing & App Notarization 

### PR DESCRIPTION
**Pull request by ImmaZoni**
## Description

Added required environmental variables to GitHub secrets and configured via the publish action. 

## Motivation and Context

This is required to prevent security popups when trying to install and run the program on OSX, it also provides a larger sense of security to a user. 

## How Has This Been Tested?

Publish action has been tested on code-sign branch, released .dmg prereleases have been tested on my work MacBook air, and on a coworkers MacBook device aswell. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by nazar-pc** [comment]
Please resolve merge conflict first

**Comment by jdheeter** [comment]
In main.yml it's using nodejs 12 but for dev I'm using 14 which is the current LTS release. I assume that github actions supports 14 so I suggest we change the action to use 14.

**Comment by jdheeter** [comment]
Resolved merge conflict

**Comment by nazar-pc** [comment]
GitHub Actions supports even Node.js 17. Let's use 14 everywhere or even 16 if it is not a problem for you (it will become LTS very soon).

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -29,16 +29,15 @@ jobs:
...
+        APPLE_CERTIFICATE_PASSWORD: ${{ secrets.MACOS_CERTIFICATE_PWD }}
+        APPLE_SIGNING_IDENTITY: ${{ secrets.MACOS_IDENTITY_ID }}
+        APPLE_ID: ${{ secrets.APPLE_ID }}
+        APPLE_PASSWORD: ${{ secrets.APPLE_PASSWORD }}
+      with:
+        tagName: app-v__VERSION__ # the action automatically replaces \_\_VERSION\_\_ with the app version
+        releaseName: "App v__VERSION__"
+        body: "See the assets to download this version and install."
+        draft: true
+        prerelease: true
```
Comment (nazar-pc) [inline .github/workflows/main.yml:43]: Please configure IDE to insert new line at the end of the file

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "subspace-desktop",
-  "version": "0.0.2",
+  "version": "0.0.2.1",
```
Comment (nazar-pc) [inline package.json:3]: Let's use semantic versioning, this should become `0.0.3`

**Review by nazar-pc — COMMENTED** [review]
I have configured GitHub Actions to require approval every time before it runs for external contributors, just to make sure we don't leak secrets when someone modifies workflow.

**Comment by nazar-pc** [comment]
Can we have any CI to check commits before merging BTW?

**Comment by ImmaZoni** [comment]
Added CI-Test in actions, had to commit directly to main to create the new action (branches act a little funky with workflows from what I've found.) 

This will test anytime there is a push, or PR

**Comment by nazar-pc** [comment]
Resolved comments do not seem to be resolved

**Comment by nazar-pc** [comment]
@ImmaZoni, the button was supposed to be green before you merge things.
And you are not supposed to push changes to `main` either, I specifically configured rules that prevent that.
Please don't make me enforce it for administrators too.

**Comment by ImmaZoni** [comment]
@nazar-pc Yes, this was my mistake, I did not intend to merge I apologize. 

somehow got clicked when I was tabbing back and forth I will ensure this doesn't happen moving forward.

**Pull request merged**